app/javascript/src/components/Dashboard/Notes/MenuBar.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from "react";
+
+import { Search, Settings, Plus } from "neetoicons";
+import { Typography } from "neetoui";
+import { MenuBar as NeetoMenuBar } from "neetoui/layouts";
+
+import { MENUBAR_BLOCKS } from "./constants";
+
+const MenuBar = () => {
+  const [isSearchCollapsed, setIsSearchCollapsed] = useState(true);
+  return (
+    <div className="flex">
+      <NeetoMenuBar showMenu title="Notes">
+        {MENUBAR_BLOCKS.main.map(menubar_block => (
+          <NeetoMenuBar.Block
+            active={menubar_block.active}
+            count={menubar_block.count}
+            key={menubar_block.label}
+            label={menubar_block.label}
+          />
+        ))}
+        <NeetoMenuBar.SubTitle
+          iconProps={[
+            {
+              icon: Search,
+              onClick: () => setIsSearchCollapsed(!isSearchCollapsed),
```

Comment:
Carefully read https://academy.bigbinary.com/learn-react/pages/know-when-to-use-previous-state-for-state-update